### PR TITLE
Add support to unicode file name

### PR DESCRIPTION
**Pull request by howl-anderson**

### CODE REVIEW

**Comment by howl-anderson** [comment]
I have updated test case and add a demo to `tiny_example.py`. If there is something else that needs I do, please tell me. I am very pleased to contribute.

File: examples/tiny_example.py [hunk 1]
```diff
@@ -34,6 +35,18 @@ def export_records():
                                           file_name="export_data")
 
 
+@app.route("/download_unicode_file_name_utf8", methods=['GET'])
```
Comment (chfw) [inline examples/tiny_example.py]: I see that you would like to show both str and unicode are handled with your change. However, I would suggest let's keep one of the two demo functions.

File: tests/test_upload_n_download_excel.py [hunk 2]
```diff
@@ -71,8 +76,11 @@ def test_no_file_type(self):
     def test_override_file_name(self):
         for file_type in FILE_TYPE_MIME_TABLE.keys():
             file_name = 'override_file_name'
+            url_encoded_file_name = quote(file_name)
```
Comment (chfw) [inline tests/test_upload_n_download_excel.py]: As you have tried to demonstrate in the tiny_example.py, could you please invest a bit more effort in writing two additional test case for the previous two demo functions.

**Comment by chfw** [comment]
It looks good so far. Please see my comments.

In terms of release, I think I will quickly get it out of the door if all required is done.

**Comment by howl-anderson** [comment]
WOW! Thank you! If there are something need improved, please let me know.

**Comment by chfw** [comment]
It will be released under New BSD license and your change will be migrated to its siblings: django-excel and pyramid-excel. Are you happy with what will happen next?

**Comment by howl-anderson** [comment]
It's a pleasure to see my code can serve so many great projects. I am very happy with that.

**Comment by chfw** [comment]
That's what I'd like to hear. Thanks. The release will follow.

**Comment by chfw** [comment]
published.

**Pull request merged**